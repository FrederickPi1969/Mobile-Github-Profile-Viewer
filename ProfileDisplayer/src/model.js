/*eslint-disable */
import { TOKEN } from '@env';  // eslint won't be able to find module @env
/* eslint-enable */

const nodeFetch = require('node-fetch');

const QUERY = `{
    user(login: "{:USER_LOGIN:}") {
      login
      name
      email
      avatarUrl
      bio
      websiteUrl
      createdAt
      updatedAt
      following(first: 100) {
        totalCount
        edges {
          node {
            login
            name
            avatarUrl
          }
        }
      }
      followers(first: 100) {
        edges {
          node {
            login
            name
            avatarUrl
          }
        }
      }
      repositories(first: 100) {
        edges {
          node {
            url
            name
            owner {
              login
            }
            description
          }
        }
      }
    }
}`;

/**
 * The Github user object, or data structure,
 * for storing users' information per requirement of assignment-3.0.
 */
export class GitUser {
  constructor(json) {
    this.parseResult(json);
  }

  parseResult(json) {
    const userData = json.data.user;
    this.userName = userData.name;
    this.userLogin = userData.login;
    this.email = userData.email;
    this.avatar = userData.avatarUrl;
    this.bio = userData.bio;
    this.createdAt = userData.createdAt;
    this.updatedAt = userData.updatedAt;
    this.followers = userData.followers;
    this.following = userData.following;
    this.repo = userData.repositories;
    this.website = userData.websiteUrl;
  }
}

/**
 * The workhorse function for the whole model in MVC.
 * It handles the core functionality of fetch user data
 * via sending post requests to github graphQL.
 * @param {String} userLogin query string, or the target github username (aka user login)
 * @param {String} AUTH authorization string to use graphql.
 * @returns
 */
const fetchData = async function
fetchData(userLogin, fetchCallback = nodeFetch, AUTH = TOKEN) {
  const url = 'https://api.github.com/graphql';
  const query = QUERY.replace(/\n/g, ' ').replace('{:USER_LOGIN:}', userLogin);
  const queryData = JSON.stringify({ query });
  let user;
  try {
    await fetchCallback(url, {
      method: 'post',
      headers: {
        'Content-Type': 'application/json',
        Accept: 'application/json',
        Authorization: `bearer ${AUTH}`,
      },
      body: queryData,
    }).then((response) => response.json()).then((data) => {
      if ('data' in data) {
        user = ('errors' in data) ? null : new GitUser(data);
      } else {
        // authorization is wrong
        user = null;
      }
    });
  } catch {
    // internet error
    user = null;
  }
  return user;
};

export const GIT_TOKEN = TOKEN;
export { fetchData };
