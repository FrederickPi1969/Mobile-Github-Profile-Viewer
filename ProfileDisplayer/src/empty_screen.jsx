/* eslint-disable global-require */
import React from 'react';
import {
  StyleSheet, Text, View, Image, TouchableOpacity,
} from 'react-native';

/**
 * Empty Screen for cases where no results
 * (including search results, repos, followers, following) are available.
 * Prompt users correspondingly.
 */
class EmptyScreen extends React.Component {
  styles = StyleSheet.create({
    back: {
      color: '#2902eb',
      fontSize: 18,
    },
    img: {
      width: '60vw',
      height: '60vw',
      marginLeft: 'auto',
      marginRight: 'auto',
      marginTop: '25%',
    },

    prompt: {
      marginTop: '2vh',
      fontWeight: 'bold',
      fontSize: 20,
      textAlign: 'center',
      fontFamily: 'Comic Sans MS',
    },
  })

  constructor(props) {
    super(props);
    const { navigation, route } = this.props;
    this.navigation = navigation;
    this.prompt = route.params.emptyPrompt;
  }

  render() {
    return (
      <View style={{ backgroundColor: '#ffb8ce', height: '100vh' }}>
        <TouchableOpacity
          onPress={() => {
            this.navigation.goBack();
            if (this.prompt.startsWith('Sorry, this GitHub user')) {
              this.navigation.goBack();
            }
          }}
        >
          <Text style={this.styles.back}>
            {'< Back'}
          </Text>
        </TouchableOpacity>

        <Image
          source={require('../assets/empty.png')}
          style={this.styles.img}
        />

        <Text style={this.styles.prompt}>
          {this.prompt}
        </Text>
      </View>
    );
  }
}

export default EmptyScreen;
/* eslint-disable global-require */
