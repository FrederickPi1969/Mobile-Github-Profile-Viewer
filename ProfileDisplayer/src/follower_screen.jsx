/* eslint-disable global-require */
import React from 'react';
import {
  StyleSheet, Text, View, TouchableOpacity,
  ImageBackground, Image,
} from 'react-native';
import { StackActions } from '@react-navigation/native';
import { fetchData } from './model';
import EmptyScreen from './empty_screen';

/**
 * Check whether the data of a certain field
 * responded by github graphql is null.
 * If yes, replace them with "not filled"
 * for displaying concern.
 * @param {String} field a field of userData to be checked with
 * @returns the field value if not null, "not filled" otherwise.
 */
const replaceNullField = function replaceNullField(field) {
  const output = (field === null || field === '') ? '' : field;
  return output;
};

/**
 * Follower Screen for demonstrating the all followers
 * the queried user.
 */
class FollowerScreen extends React.Component {
  styles = StyleSheet.create({
    back: {
      color: '#2902eb',
      fontSize: 18,
    },

    header: {
      marginTop: '5%',
      marginLeft: '3%',
      fontSize: 20,
      fontWeight: 'bold',
    },

    followerWrapper: {
      width: '85%',
      marginLeft: 'auto',
      marginRight: 'auto',
      marginTop: '2.5vh',
      borderRadius: 8,
      borderColor: 'black',
      borderStyle: 'solid',
      borderWidth: 1,
      backgroundColor: '#e6e4e3',
      paddingBottom: '0.5vh',
    },

    avatar: {
      borderRadius: 500,
      width: '9vh',
      height: '9vh',
    },

    followerLogin: {
      fontWeight: 'bold',
      fontSize: 20,
      fontFamily: "'Comic Sans MS'",
      color: '#2a05e3',
    },

    followerName: {
      fontSize: 16,
      fontWeight: '400',
      color: '#ff8800',
    },
  })

  constructor(props) {
    super(props);
    const { navigation, route } = this.props;
    this.navigation = navigation;
    this.followers = route.params.followers;
    this.user = route.params.user;
  }

  /**
   * When user clicked on follower,
   * navigate to loading page first &
   * send post request to graphql,
   * and redirect user to the follower profile page
   * when data get ready.
   * @param {string} followerLogin the login of follower being clicked
   */
  async handleNav(followerLogin) {
    const pushLoading = StackActions.push('Loading');
    this.navigation.dispatch(pushLoading);
    // this.navigation.navigate('Loading');
    await fetchData(followerLogin)
      .then((userData) => {
        if (userData === null) {
        // error - non existing user
          const emptyPrompt = 'Sorry, this GitHub user no long exists';
          const pushEmpty = StackActions.push('Empty', { emptyPrompt });
          this.navigation.dispatch(pushEmpty);
        } else {
          const pushProfile = StackActions.push('Profile', { userData });
          this.navigation.dispatch(pushProfile);
        }
      });
  }

  /**
   * Add all components (viewer wrappers) containing
   * the users are following the current queried user to screen.
   * @returns a list of react-native components to be directly rendered.
   */
  addAllFollowers() {
    const items = [];
    for (let i = 0; i < this.followers.length; i += 1) {
      const follower = this.followers[i].node;
      const followerAvatarURL = follower.avatarUrl;
      const followerLogin = follower.login;
      const followerName = follower.name;
      items.push(
        <TouchableOpacity
          style={this.styles.followerWrapper}
          key={`Follower${i}`}
          testID={followerLogin}
          onPress={() => { this.handleNav(followerLogin); }}
        >
          <View
            style={{ marginLeft: '1.5vw', marginTop: '0.5vh', flexDirection: 'row' }}
            testID={`FollowerImgWrapper${i}`}
          >
            <Image
              testID={`FollowerAvatar${i}`}
              source={{ uri: followerAvatarURL }}
              style={this.styles.avatar}
            />
            <View
              testID={`FollowerTextWrapper${i}`}
              style={{ marginLeft: '6.5vw', marginTop: '1vh' }}
            >
              <Text testID={`FollowerLogin${i}`} style={this.styles.followerLogin}>
                {followerLogin}
              </Text>
              <Text testID={`FollowerName${i}`} style={this.styles.followerName}>
                {replaceNullField(followerName)}
              </Text>
            </View>
          </View>
        </TouchableOpacity>,
      );
    }
    return items;
  }

  render() {
    if (this.followers === null || this.followers === undefined
        || this.followers.length === 0) {
      // this is only for testing and should never be reached otherwise
      // This checking is done on profile screen when handling navigation,
      // and this condition will never be true in practice usage.
      return (
        <EmptyScreen
          testID="Empty"
          navigation={null}
          route={{ params: { emptyPrompt: 'Empty!' } }}
        />
      );
    }

    return (
      <View>
        <ImageBackground
          testID="background"
          style={{ minHeight: '100vh' }}
          source={require('../assets/home_background.jpg')}
        >
          <TouchableOpacity
            testID="back"
            onPress={() => this.navigation.goBack()}
          >
            <Text style={this.styles.back} testID="backString">
              {'< Back'}
            </Text>
          </TouchableOpacity>

          <Text style={this.styles.header} testID="header">
            {`${this.user}'s Followers`}
          </Text>

          <View style={{ marginTop: '5%' }} testID="Wrapper">
            {this.addAllFollowers()}
          </View>
        </ImageBackground>
      </View>
    );
  }
}

export default FollowerScreen;
/* eslint-disable global-require */
