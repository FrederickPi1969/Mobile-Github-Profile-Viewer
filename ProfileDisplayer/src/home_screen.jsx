/* eslint-disable global-require */
import React from 'react';
import {
  StyleSheet, Text, View, TextInput,
  TouchableOpacity, Image, ImageBackground,
} from 'react-native';
import { StackActions } from '@react-navigation/native';
import { fetchData } from './model';

/**
 * Home Page of the app. Enable the users to query
 * github users per github login.
 */
class HomeScreen extends React.Component {
    styles = StyleSheet.create({
      logo: {
        width: '33vw',
        height: '33vw',
        borderRadius: 500,
        marginLeft: 'auto',
        marginRight: 'auto',
        marginTop: '25%',

      },
      header: {
        textAlign: 'center',
        fontWeight: 'bold',
        marginTop: '15%',
        fontSize: 20,
      },
      inputBox: {
        fontSize: 18,
        borderWidth: 2,
        borderStyle: 'solid',
        borderColor: 'black',
        color: '#ebebeb',
        borderRadius: 5,
        marginTop: '8%',
        textAlign: 'center',
        marginLeft: 'auto',
        marginRight: 'auto',
        width: '80%',
        height: '5.5vh',
      },
      submit: {
        color: 'green',
        backgroundColor: '#4a4a4a',
        marginTop: '10%',
        borderWidth: 0.5,
        borderStyle: 'solid',
        borderColor: 'black',
        // border: '0.5px solid black',
        borderRadius: 8,
        width: '25%',
        height: '5.5vh',
        marginLeft: '37.5%',
        textAlign: 'center',
      },
    });

    constructor(props) {
      super(props);
      this.state = { targetUser: 'BillyZhaohengLi' };
    }

    /**
     * Fetch target user via graphQL API of github.
     * Then first navigate to Loading Screen.
     * After fetching is ready, navigate to Profile Screen.
     */
    getTargetUser() {
      const { navigation } = this.props;
      const { targetUser } = this.state;
      const pushLoading = StackActions.push('Loading');
      navigation.dispatch(pushLoading);
      fetchData(targetUser)
        .then((userData) => {
          if (userData === null) {
            // error - non existing user
            const emptyPrompt = 'Sorry, this GitHub user does not exist, or'
                                    + ' your authorization token is wrong.';
            const pushEmpty = StackActions.push('Empty', { emptyPrompt });
            navigation.dispatch(pushEmpty);
          } else {
            const pushProfile = StackActions.push('Profile', { userData });
            navigation.dispatch(pushProfile);
          }
        });
    }

    render() {
      const { targetUser } = this.state;
      return (
        <ImageBackground
          source={require('../assets/home_background.jpg')}
          style={{ minHeight: '100vh' }}
        >
          <View>
            <Image
              source={require('../assets/git_logo.jpg')}
              style={this.styles.logo}
            />
            <Text style={this.styles.header}>Which GitHub User Are you Looking for?</Text>
            <TextInput
              style={this.styles.inputBox}
              onChangeText={(text) => this.setState({ targetUser: text })}
              defaultValue={targetUser}
            />

            <TouchableOpacity
              onPress={() => this.getTargetUser()}
              style={this.styles.submit}
            >
              <Text style={{
                fontWeight: 'bold',
                textAlign: 'center',
                color: '#ebebeb',
                marginTop: 'auto',
                marginBottom: 'auto',
              }}
              >
                Search
              </Text>
            </TouchableOpacity>
          </View>
        </ImageBackground>

      );
    }
}

export default HomeScreen;
/* eslint-disable global-require */
