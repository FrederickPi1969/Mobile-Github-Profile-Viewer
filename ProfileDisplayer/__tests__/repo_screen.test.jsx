import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import RepoScreen from '../src/repo_screen';
import { fetchData } from '../src/model';

const fetch = require('jest-fetch-mock');

beforeEach(() => {
  fetch.resetMocks();
});

it('renders correctly - good condition', async () => {
  await fetchData('FrederickPi1969')
    .then((userData) => ({
      params: {
        repo: userData.repo.edges,
        user: userData.userLogin,
      },
    }))
    .then((params) => {
      const tree = renderer.create(<RepoScreen
        navigation={null}
        route={params}
      />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});

it('renders correctly - user has 0 repos', async () => {
  await fetchData('FrederickPi1969')
    // Real bad condition - user has no public repos.
    .then((userData) => ({
      params: {
        repo: [],
        user: userData.userLogin,
      },
    }))
    .then((params) => {
      const tree = renderer.create(<RepoScreen
        navigation={null}
        route={params}
      />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});

it('renders correctly - mocked network error', async () => {
  await fetchData('FrederickPi1969', fetch) // using mocked fetch!
  // Mocked bad condition - network error
    .then((userData) => {
      if (userData !== null) {
        throw new Error('Error!');
      }
      return {
        params: {
          repo: null,
          user: null,
        },
      };
    })
    .then((params) => {
      const tree = renderer.create(<RepoScreen
        navigation={null}
        route={params}
      />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});

it('renders correctly - mocked other errors', async () => {
  // Mocked bad condition - unexpected errors, such as
  // repo page permanently removed, or authorization error
  fetch.mockResponseOnce(JSON.stringify({ data: { error: 'unexpected error.' } }));
  await fetchData('FrederickPi1969', fetch) // using mocked fetch!
    .then((userData) => {
      if (userData !== null) {
        throw new Error('Error!');
      }
      return {
        params: {
          repo: null,
          user: null,
        },
      };
    })
    .then((params) => {
      const tree = renderer.create(<RepoScreen
        navigation={null}
        route={params}
      />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});
