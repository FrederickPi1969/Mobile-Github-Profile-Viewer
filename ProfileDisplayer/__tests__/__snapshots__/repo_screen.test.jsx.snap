// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly - good condition 1`] = `
<View>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "minHeight": "100vh",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/home_background.jpg",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#2902eb",
            "fontSize": 18,
          }
        }
      >
        &lt; Back
      </Text>
    </View>
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
          "marginLeft": "3%",
          "marginTop": "5%",
        }
      }
    >
      FrederickPi1969's Repos
    </Text>
    <View
      style={
        Object {
          "marginTop": "5%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#d5ccff",
            "borderColor": "black",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": "2%",
            "width": "85%",
          }
        }
      >
        <View
          style={
            Object {
              "marginLeft": "1.5vw",
              "marginTop": "0.5vh",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#2a05e3",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Course-Projects
          </Text>
          <Text
            style={
              Object {
                "color": "#626166",
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }
          >
            Not filled
          </Text>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "color": "#ff8800",
                "fontSize": 12,
                "fontWeight": "bold",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ff8800",
                  "fontSize": 12,
                  "fontWeight": "bold",
                }
              }
            >
              FrederickPi1969
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#d5ccff",
            "borderColor": "black",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": "2%",
            "width": "85%",
          }
        }
      >
        <View
          style={
            Object {
              "marginLeft": "1.5vw",
              "marginTop": "0.5vh",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#2a05e3",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            iRM_Autonomy_2020
          </Text>
          <Text
            style={
              Object {
                "color": "#626166",
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }
          >
            Awesome Repo for iRM 2020 CV Development
          </Text>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "color": "#ff8800",
                "fontSize": 12,
                "fontWeight": "bold",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ff8800",
                  "fontSize": 12,
                  "fontWeight": "bold",
                }
              }
            >
              illini-robomaster
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#d5ccff",
            "borderColor": "black",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": "2%",
            "width": "85%",
          }
        }
      >
        <View
          style={
            Object {
              "marginLeft": "1.5vw",
              "marginTop": "0.5vh",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#2a05e3",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Taxnonomy
          </Text>
          <Text
            style={
              Object {
                "color": "#626166",
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }
          >
            Not filled
          </Text>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "color": "#ff8800",
                "fontSize": 12,
                "fontWeight": "bold",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ff8800",
                  "fontSize": 12,
                  "fontWeight": "bold",
                }
              }
            >
              FrederickPi1969
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#d5ccff",
            "borderColor": "black",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": "2%",
            "width": "85%",
          }
        }
      >
        <View
          style={
            Object {
              "marginLeft": "1.5vw",
              "marginTop": "0.5vh",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#2a05e3",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            CS512-CGG
          </Text>
          <Text
            style={
              Object {
                "color": "#626166",
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }
          >
            CGG Paper
          </Text>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "color": "#ff8800",
                "fontSize": 12,
                "fontWeight": "bold",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ff8800",
                  "fontSize": 12,
                  "fontWeight": "bold",
                }
              }
            >
              FrederickPi1969
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#d5ccff",
            "borderColor": "black",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": "2%",
            "width": "85%",
          }
        }
      >
        <View
          style={
            Object {
              "marginLeft": "1.5vw",
              "marginTop": "0.5vh",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#2a05e3",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Subteam-replacement-svn
          </Text>
          <Text
            style={
              Object {
                "color": "#626166",
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }
          >
            Not filled
          </Text>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "color": "#ff8800",
                "fontSize": 12,
                "fontWeight": "bold",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ff8800",
                  "fontSize": 12,
                  "fontWeight": "bold",
                }
              }
            >
              BillyZhaohengLi
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#d5ccff",
            "borderColor": "black",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": "2%",
            "width": "85%",
          }
        }
      >
        <View
          style={
            Object {
              "marginLeft": "1.5vw",
              "marginTop": "0.5vh",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#2a05e3",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            CS241
          </Text>
          <Text
            style={
              Object {
                "color": "#626166",
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }
          >
            Not filled
          </Text>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "color": "#ff8800",
                "fontSize": 12,
                "fontWeight": "bold",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ff8800",
                  "fontSize": 12,
                  "fontWeight": "bold",
                }
              }
            >
              FrederickPi1969
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#d5ccff",
            "borderColor": "black",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": "2%",
            "width": "85%",
          }
        }
      >
        <View
          style={
            Object {
              "marginLeft": "1.5vw",
              "marginTop": "0.5vh",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#2a05e3",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            frederickpi.github.io
          </Text>
          <Text
            style={
              Object {
                "color": "#626166",
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }
          >
            Not filled
          </Text>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "color": "#ff8800",
                "fontSize": 12,
                "fontWeight": "bold",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ff8800",
                  "fontSize": 12,
                  "fontWeight": "bold",
                }
              }
            >
              FrederickPi1969
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#d5ccff",
            "borderColor": "black",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": "2%",
            "width": "85%",
          }
        }
      >
        <View
          style={
            Object {
              "marginLeft": "1.5vw",
              "marginTop": "0.5vh",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#2a05e3",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            subteam_replacement_paper
          </Text>
          <Text
            style={
              Object {
                "color": "#626166",
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }
          >
            Code repository for the arXiv preprint at arxiv.org/abs/2101.11070
          </Text>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "color": "#ff8800",
                "fontSize": 12,
                "fontWeight": "bold",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ff8800",
                  "fontSize": 12,
                  "fontWeight": "bold",
                }
              }
            >
              BillyZhaohengLi
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#d5ccff",
            "borderColor": "black",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": "2%",
            "width": "85%",
          }
        }
      >
        <View
          style={
            Object {
              "marginLeft": "1.5vw",
              "marginTop": "0.5vh",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#2a05e3",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Bili-Impression
          </Text>
          <Text
            style={
              Object {
                "color": "#626166",
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }
          >
            Not filled
          </Text>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "color": "#ff8800",
                "fontSize": 12,
                "fontWeight": "bold",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ff8800",
                  "fontSize": 12,
                  "fontWeight": "bold",
                }
              }
            >
              FrederickPi1969
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly - mocked network error 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffb8ce",
      "height": "100vh",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#2902eb",
          "fontSize": 18,
        }
      }
    >
      &lt; Back
    </Text>
  </View>
  <Image
    source={
      Object {
        "testUri": "../../../assets/empty.png",
      }
    }
    style={
      Object {
        "height": "60vw",
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": "25%",
        "width": "60vw",
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "Comic Sans MS",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginTop": "2vh",
        "textAlign": "center",
      }
    }
  >
    Empty!
  </Text>
</View>
`;

exports[`renders correctly - mocked other errors 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffb8ce",
      "height": "100vh",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#2902eb",
          "fontSize": 18,
        }
      }
    >
      &lt; Back
    </Text>
  </View>
  <Image
    source={
      Object {
        "testUri": "../../../assets/empty.png",
      }
    }
    style={
      Object {
        "height": "60vw",
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": "25%",
        "width": "60vw",
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "Comic Sans MS",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginTop": "2vh",
        "textAlign": "center",
      }
    }
  >
    Empty!
  </Text>
</View>
`;

exports[`renders correctly - user has 0 repos 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffb8ce",
      "height": "100vh",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#2902eb",
          "fontSize": 18,
        }
      }
    >
      &lt; Back
    </Text>
  </View>
  <Image
    source={
      Object {
        "testUri": "../../../assets/empty.png",
      }
    }
    style={
      Object {
        "height": "60vw",
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": "25%",
        "width": "60vw",
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "Comic Sans MS",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginTop": "2vh",
        "textAlign": "center",
      }
    }
  >
    Empty!
  </Text>
</View>
`;
